examples/sealed-auctions/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+                    item_id
+                ));
+            }
+            let mut auction: &mut Auction = self.auctions.get_mut(&item_id).unwrap();
+            if auction.seller == ctx.sender() {
+                return Err(failure::format_err!(
+                    "The seller cannot also be a bidder for an item."
+                ));
+            }
+            if self.blacklist.contains(&ctx.sender()) {
+                return Err(failure::format_err!(
+                    "You don't have permission to bid. Please contact the auction owner."
+                ));
+            }
+            if value < auction.reserve {
+                return Err(failure::format_err!(
+                    "The bid has to be higher than the reserve of {}.",
+                    auction.reserve
+                ));
+            }
+            if value < auction.max_bid {
+                return Err(failure::format_err!(
+                    "The bid is lower than the current maximum."
+                ));
+            }
+
+            // Create/update bid
+            if !auction.bids.contains(&ctx.sender()) {
+                auction.bids.insert(Bid {
+                    bidder: ctx.sender(),
+                    value: value,
+                });
+            } else {
+                let mut bid = auction.bids.take(&ctx.sender()).unwrap();
+                if bid.value > value || value < auction.max_bid {
+                    return Err(failure::format_err!(
+                        "A new bid has to be greater than {}.",
+                        std::cmp::max(bid.value, auction.max_bid)
+                    ));
+                }
+                bid.value = value;
+                auction.bids.insert(bid);
+            }
+            auction.max_bid = value;
+
+            Ok(())
+        }
+
+        /// Close an auction
+        /// Only the seller can close an auction
+        pub fn close_auction(&mut self, ctx: &Context, item_id: ItemId) -> Result<Auction> {
+            if !self.auctions.contains_key(&item_id) {
+                return Err(failure::format_err!(
+                    "There is no active auction for item {}.",
+                    item_id
+                ));
+            }
+            let mut auction = self.auctions.remove(&item_id).unwrap();
+            if auction.seller != ctx.sender() {
+                return Err(failure::format_err!(
+                    "The seller is the only one who can close an auction."
+                ));
+            }
+            if auction.bids.len() == 0 {
+                return Err(failure::format_err!(
+                    "There are no bids on item {}. Closing auction.",
+                    item_id
+                ));
+            }
+
+            // get artifact
+            let mut artifact: &mut Artifact = self.artifacts.get_mut(&item_id).unwrap();
+
+            // sort the bids and pick the winner and price
+            let mut bidsVec = Vec::new();
```

Comment:
heh heh looks like some other language snuck in here 😉 
```suggestion
            let mut bids_vec = Vec::new();
```

Reply:
but actually the following is the more idiomatic way to create a `Vec` from some other collection
```suggestion
            let mut bids_vec: Vec<_> = auction.bids.iter().collect();
```

Reply:
Done